src/motion_input_shaping/shaper_config.py

```diff
@@ -0,0 +1,84 @@
+"""This file contains the ShaperConfig class which is used to specify shaping mode and settings."""
+
+import warnings
+from enum import Enum
+from typing import Any
+from zero_vibration_stream_generator import ShaperType
+
+
+class ShaperMode(Enum):
+    """Enumeration for different shaper modes."""
+
+    DECEL = 1
+    STREAM = 2
+
+
+class Settings(dict[str, Any]):
+    """Settings dictionary where keys and value types are fixed after initialization."""
+
+    def __setitem__(self, key: str, value: Any) -> None:
+        """Set item in dictionary if it matches an existing item."""
+        if key not in self:
+            raise KeyError(f"{key} is not a setting.")
+
+        if not isinstance(value, type(self[key])):
+            # be strict about the type of values
+            raise KeyError(f"Value provided for {key} must by of type {type(self[key])}.")
+
+        super().__setitem__(key, value)
+
+
+class ShaperConfig:
+    """Configuration to class to select a shaper mode and specify settings."""
+
+    def __init__(self, shaper_mode: ShaperMode, **options: Any):
+        """
+        Initialize the class in specified mode and with options.
+
+        :param ShaperMode: Method to use to create shaping
+        :param options: Settings specified as keyword pairs
+        """
+        self.shaper_mode = shaper_mode
+        self._write_settings(options)  # process key word arguments and
```

Comment:
and?

Reply:
This class has been removed so this is no longer a problem